cognite/neat/utils/utils.py

```diff
@@ -75,15 +87,25 @@ def _get_cognite_client(config: CogniteClientConfig, credentials: CredentialProv
 
 
 @overload
-def remove_namespace_from_uri(*URI: URIRef | str, special_separator: str = "#_") -> str: ...
+def remove_namespace_from_uri(
+    *URI: URIRef | str,
+    special_separator: str = "#_",
+    deep_validation: bool = False,
+) -> str: ...
 
 
 @overload
-def remove_namespace_from_uri(*URI: tuple[URIRef | str, ...], special_separator: str = "#_") -> tuple[str, ...]: ...
+def remove_namespace_from_uri(
+    *URI: tuple[URIRef | str, ...],
+    special_separator: str = "#_",
+    deep_validation: bool = False,
+) -> tuple[str, ...]: ...
 
 
 def remove_namespace_from_uri(
-    *URI: URIRef | str | tuple[URIRef | str, ...], special_separator: str = "#_"
+    *URI: URIRef | str | tuple[URIRef | str, ...],
+    special_separator: str = "#_",
+    deep_validation: bool = False,
```

Comment:
I think it is good to avoid flags if we can.
What about making this a Literal and then in the docs explain the difference. 
```suggestion
    validation: Literal["full", "prefix"] = "prefix",
```
```python
"""
validation: How to validate the URI. Full checks that it is a valid HttpURL, while prefix that it starts with 'http'
"""
```